src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -1103,42 +1102,34 @@ public void updateInstallation(int id, String json) {
     }
 
     private Request req(String method, Uri template) {
-        RequestBuilder builder = new RequestBuilder(method);
-        if (realm != null) {
-            builder.setRealm(realm);
-        } else {
-            builder.addHeader("Authorization", "Bearer " + oauthToken);
-        } 
-        builder.setUrl(template.toString());
-        return builder.build();
+        return req(method, template.toString());
     }
 
-    private Request req(String method, Uri template, String contentType, byte[] body) {
+    private static final Pattern RESTRICTED_PATTERN = Pattern.compile("%2B", Pattern.LITERAL);
+
+    private Request req(String method, String url) {
         RequestBuilder builder = new RequestBuilder(method);
         if (realm != null) {
             builder.setRealm(realm);
         } else {
             builder.addHeader("Authorization", "Bearer " + oauthToken);
-        } 
-        builder.setUrl(template.toString());
-        builder.addHeader("Content-type", contentType);
-        builder.setBody(body);
+        }
+        builder.setUrl(RESTRICTED_PATTERN.matcher(url).replaceAll("+")); // replace out %2B with + due to API restriction
         return builder.build();
     }
 
-    private Request req(String method, Uri template, int page) {
+    private Request req(String method, Uri template, String contentType, byte[] body) {
         RequestBuilder builder = new RequestBuilder(method);
         if (realm != null) {
             builder.setRealm(realm);
         } else {
             builder.addHeader("Authorization", "Bearer " + oauthToken);
         }
-        builder.addQueryParameter("page", Integer.toString(page));
-        builder.setUrl(template.toString().replace("%2B", "+")); //replace out %2B with + due to API restriction
```

Comment:
optimized API restriction replacement as well, now it doesn't compile a new pattern every request.